OCM localization support

mandelsoft opened the pull request:
**What this PR does / why we need it**:

This PR provides some support for localization and configuration of filesystem snapshots
based on information provided by component versions and user configuration.

Additionally, it contains a fix for the Tar resource generation.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       breaking|feature|bugfix|doc|other
- target_group:   user|operator|developer|dependency
-->
```feature user

```

mandelsoft committed:
OCM localization support

mandelsoft committed:
fix lint problem

mandelsoft committed:
fix used yaml parser

mandelsoft committed:
CLI: fix tar input handling

mandelsoft committed:
use yamlpath for substitutions

Skarlso reviewed (changes requested):
Some change requests in comments above.

mandelsoft committed:
incorporate review

Additionally I found a strange behaviour of the yaml path library.
It cannot handle mixed yaml json. The resulting output when executing
a modification is a completely corrupted yaml.

Now, the susbtitution package wraps this lib by feeding it with
pure json or yaml, only, dependening on the original input format.

mandelsoft committed:
lint errors and review
